File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/WebviewCreateWebviewPanelOptions.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class WebviewCreateWebviewPanelOptions(
  val enableScripts: Boolean,
  val enableForms: Boolean,
  val enableOnlyCommandUris: List<String>? = null,
  val localResourceRoots: List<String>? = null,
  val portMapping: List<PortMappingParams>,
  val enableFindWidget: Boolean,
  val retainContextWhenHidden: Boolean,
)

File: agent/bindings/csharp/WebviewCreateWebviewPanelOptions.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class WebviewCreateWebviewPanelOptions
  {

    [JsonPropertyName("enableScripts")]
    public bool EnableScripts { get; set; }

    [JsonPropertyName("enableForms")]
    public bool EnableForms { get; set; }

    [JsonPropertyName("enableOnlyCommandUris")]
    public string[] EnableOnlyCommandUris { get; set; }

    [JsonPropertyName("localResourceRoots")]
    public string[] LocalResourceRoots { get; set; }

    [JsonPropertyName("portMapping")]
    public PortMappingParams[] PortMapping { get; set; }

    [JsonPropertyName("enableFindWidget")]
    public bool EnableFindWidget { get; set; }

    [JsonPropertyName("retainContextWhenHidden")]
    public bool RetainContextWhenHidden { get; set; }
  }
}
